runtime/src/main/java/io/quarkiverse/resteasy/problem/validation/ConstraintViolationExceptionMapper.java

```diff
@@ -141,7 +141,10 @@ private String serializePath(Path propertyPath, int skipFirstSegments) {
                 skipFirstSegments -= 1;
                 segmentIterator.next();
             } else {
-                pathSegments.add(segmentIterator.next().toString());
+                String segment = segmentIterator.next().toString();
+                if (segment != null && !segment.trim().isEmpty()) {
+                    pathSegments.add(segment);
+                }
```

Comment:
I think it's a bit too late to check for segment nullability, IMO it never can be null, because `segmentIterator.next()` never returns null thanks to `hasNext()` check above.

also, `segment.trim().isEmpty()` has shorter version: `segment.isBlank()` since java 11.

```suggestion
                String segment = segmentIterator.next().toString();
                if (!segment.isBlank()) {
                    pathSegments.add(segment);
                }
```